src/components/tooltips-hoc/DropPod.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import * as _ from 'underscore';
+import {positionValidator} from './positionValidation';
+
+export const DropPodPosition = {
+    bottom: 'bottom',
+    top: 'top',
+    left: 'left',
+    right: 'right',
+};
+
+export interface IDropPodProps {
+    isOpen: boolean;
+    positions?: string[];
+    buttonRef: React.RefObject<HTMLElement>;
+    renderTooltip: (style: React.CSSProperties, containerRef: React.RefObject<HTMLElement>) => React.ReactNode;
+    selector: string;
+}
+
+export interface IDropPodState {
+    offset: ClientRect | DOMRect;
+}
+
+export class DropPod extends React.PureComponent<IDropPodProps, IDropPodState> {
+    readonly tooltip: React.RefObject<HTMLElement>;
+
+    static defaultProps: Partial<IDropPodProps> = {
+        selector: '#App',
+        isOpen: false,
+        positions: [DropPodPosition.right, DropPodPosition.bottom, DropPodPosition.top, DropPodPosition.left],
+    };
+
+    constructor(props: IDropPodProps, state: IDropPodState) {
+        super(props, state);
+
+        this.tooltip = React.createRef();
+        this.state = {
+            offset: undefined,
+        };
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.updateOffset as any);

```

Comment:
Can you remove `as any` ?

Reply:
I have a ts error about my return interface, I dont know why because its the same everywhere... If I can, I will but for now, I dont know how hahah